File: vuln-sandbox/python/taint/cmdi-2hop.py
# Taint: Command Injection — 2-hop (Python)
# Source: request.form.get('filename') → intermediate: process() → Sink: subprocess.run
# Expected: TruePositive (taint/cwe78)
from flask import Flask, request
import subprocess

app = Flask(__name__)

def process(filename):
    subprocess.run(f"convert {filename} output.png", shell=True)  # SINK

@app.route('/convert', methods=['POST'])
def convert():
    filename = request.form.get('filename')    # SOURCE: HTTP request form
    process(filename)                          # INTERMEDIATE
    return 'converted'

File: vuln-sandbox/node/taint/cmdi-2hop.js
// Taint: Command Injection — 2-hop (JS)
// Source: req.body.filename → intermediate: processFile() → Sink: exec
// Expected: TruePositive (taint/cwe78)
const express = require('express');
const { exec } = require('child_process');
const app = express();

function processFile(filename) {
    exec(`convert ${filename} output.png`);     // SINK: shell command
}

app.post('/convert', (req, res) => {
    const filename = req.body.filename;         // SOURCE: HTTP request body
    processFile(filename);                      // INTERMEDIATE
    res.send('done');
});
